app/services/catalog/cancel_order.rb

```diff
@@ -0,0 +1,32 @@
+module Catalog
+  class CancelOrder
+    UNCANCELABLE_STATES = %w[Completed Failed].freeze
+    attr_reader :order
+
+    def initialize(order_id)
+      @order = Order.find(order_id)
+    end
+
+    def process
+      raise Catalog::OrderUncancelable if UNCANCELABLE_STATES.include?(@order.state)
+
+      approval_requests.each do |approval_request|
```

Comment:
This may be a question we dealt with this week, but are we sure we want to mark all `approval_requests` as canceled? For tracking would we possibly want to mark only the approval_request canceled for the `order_item` that was canceled?

Reply:
Yeah, I'm not sure the answer to this one, because as I understand now, each order only has one order item, and each order item only has one approval request.

Reply:
@bzwei @mkanoor any thoughts here?

Reply:
@syncrou Yes that was the plan of having an order item collection inside an order, since all our code accounts for that everywhere lets not break that trend.